src/components/03-components/03-buttons/01-button/_button--single.njk

```diff
@@ -0,0 +1,68 @@
+{#
+    Creates a button.
+    link        (string) - use "a" instead of "button". String should be any valid HREF value
+    modifiers   (array)  - USA "secondary,base,outline,inverse,big,unstyled,hover,active,focus"
+    smodifiers  (array)  - SDS "circle,small"
+    disabled    (bool)   - adds disabled attribute 
+    classes     (string) - extra classes
+    ext         (string) - other attributes i.e. ( tabindex="0" onclick="alert" )
+    text        (string) - text of the button 
+    icon        (object) - icon before text if there is text {prefix,icon,title} 
+    icon_right  (object) - icon after text if there is text {prefix,icon,title} 
+#}
+<{%
```

Comment:
please keep if statement/for loop openers and closers on same line (ie: {% if link %} and {% endif %} for readability

Reply:
all white space chars outside the brackets gets passed to the output. This keeps the logic readable while ensuring a neat output

Reply:
https://mozilla.github.io/nunjucks/templating.html#whitespace-control